python/trim_osc/Portfile

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           active_variants 1.1
+PortGroup           github 1.0
+PortGroup           python 1.0
+
+github.setup        Zverik regional 7d473e463e1570bdd3b846c3d2539092c63bd98a
+version             20210417
+revision            0
+github.tarball_from archive
+
+name                trim_osc
+
+platforms           darwin
+maintainers         {@frankdean fdsd.co.uk:frank.dean} openmaintainer
+
+license             WTFPL
+
+description         Scripts for regional OSM extracts support
+
+long_description    It is hard to maintain OSM tile service with a small \
+                    server: you can't have a properly updated regional \
+                    extract or even run osm2pgsql on low memory. This \
+                    package constains some scripts that help.
+
+depends_lib-append  port:py${python.version}-lxml \
+                    port:py${python.version}-psycopg2 \
+                    port:py${python.version}-shapely
+
+checksums           rmd160  6e387ad3d613b3848778dd92df60d3c6296ac1b3 \
+                    sha256  c0eee61c14ef462d85e779bb108dedcb2054cf5ee453b2ec6a61039c0eabc648 \
+                    size    7041
+
+python.default_version  39
+
+# Database variants (from the GDAL port)
+set postgresql_suffixes {12 13}
+
+set portsgresql_variants    {}
+set postgresql_default_variant "if {"
+
+foreach s ${postgresql_suffixes} {
+    lappend portsgresql_variants postgresql${s}
+    set postgresql_default_variant "${postgresql_default_variant}!\[variant_isset postgresql${s}] && "
+}
+
+set postgresql_default_variant [string range ${postgresql_default_variant} 0 end-4]
+set postgresql_default_variant "${postgresql_default_variant}} {default_variants +postgresql${s}}"
+eval $postgresql_default_variant
+
+foreach s ${postgresql_suffixes} {
+    set p postgresql${s}
+    set v [string index ${s} 0].[string index ${s} 1]
+    set i [lsearch -exact ${portsgresql_variants} ${p}]
+    set c [lreplace ${portsgresql_variants} ${i} ${i}]
+    variant ${p} description "Enable PostgreSQL ${v} support" conflicts {*}${c} "
+        depends_lib-append      port:${p}
+    "
+}
+
+use_configure       no
+
+variant postgresql12 conflicts postgresql13 description {Use with PostgreSQL 12} {
+    require_active_variants py${python.version}-psycopg2 postgresql12
+}
+
+variant postgresql13 conflicts postgresql12 description {Use with PostgreSQL 13} {
+    require_active_variants py${python.version}-psycopg2 postgresql13
+}
```

Comment:
same comment about defining postgresql variants as in the other PR; this actually doesn't work and is overly complicated.

Reply:
Ok. Done.